rskj-core/src/test/java/co/rsk/peg/federation/FederationSupportImplTest.java

```diff
@@ -2672,6 +2681,91 @@ void voteFederationChange_add100Members_returnsSuccessResponseCodeAndFedSize100(
+                .build();
+
+            Transaction tx = TransactionUtils.getTransactionFromCaller(signatureCache, FederationChangeCaller.FIRST_AUTHORIZED.getRskAddress());
+            Transaction tx2 = TransactionUtils.getTransactionFromCaller(signatureCache, FederationChangeCaller.SECOND_AUTHORIZED.getRskAddress());
+
+            ABICallSpec createFederationAbiCallSpec = new ABICallSpec(FederationChangeFunction.CREATE.getKey(), new byte[][]{});
+
+            // Voting with  m of n authorizers to create the pending federation
+            federationSupport.voteFederationChange(tx, createFederationAbiCallSpec, signatureCache, bridgeEventLogger);
+            federationSupport.voteFederationChange(tx2, createFederationAbiCallSpec, signatureCache, bridgeEventLogger);
+
+            byte[][] commitFederationEventTopic = BridgeEvents.COMMIT_FEDERATION.getEvent().encodeEventTopics();
+            List<DataWord> encodedTopics = LogInfo.byteArrayToList(commitFederationEventTopic);
+            DataWord expectedCommitTopic = encodedTopics.get(0);
+
+            int EXPECTED_COUNT_OF_MEMBERS = 10;
+
+            // Voting add new fed with m of n authorizers
+
+            Set<String> expectedPubKeys = new HashSet<>();
+
+            for(int i = 0; i < EXPECTED_COUNT_OF_MEMBERS; i++) {
+                BtcECKey expectedBtcECKey = new BtcECKey();
+                expectedPubKeys.add(HexUtils.toJsonHex(expectedBtcECKey.getPubKey()));
+                ABICallSpec addFederationAbiCallSpec = new ABICallSpec(FederationChangeFunction.ADD.getKey(), new byte[][]{expectedBtcECKey.getPubKey()});
+                federationSupport.voteFederationChange(tx, addFederationAbiCallSpec, signatureCache, bridgeEventLogger);
+                federationSupport.voteFederationChange(tx2, addFederationAbiCallSpec, signatureCache, bridgeEventLogger);
+            }
+
+            Keccak256 pendingFederationHash = federationSupport.getPendingFederationHash();
+
+            ABICallSpec commitFederationAbiCallSpec = new ABICallSpec(FederationChangeFunction.COMMIT.getKey(), new byte[][]{pendingFederationHash.getBytes()});
+
+            Federation initialActiveFederation = federationSupport.getActiveFederation();
+
+            // Act
+
+            // Voting commit new fed with m of n authorizers
+            federationSupport.voteFederationChange(tx, commitFederationAbiCallSpec, signatureCache, bridgeEventLogger);
+            federationSupport.voteFederationChange(tx2, commitFederationAbiCallSpec, signatureCache, bridgeEventLogger);
+
+            // Assert
+
+            assertNull(federationSupport.getPendingFederationHash());
+            // -1 because the pending fed no longer exist
+            assertThat(federationSupport.getPendingFederationSize(), is(-1));
+
+            Federation finalActiveFederation = federationSupport.getActiveFederation();
+
+            assertThat(federationSupport.getActiveFederationSize(), is(EXPECTED_COUNT_OF_MEMBERS));
```

Comment:
nit: i'd move this assertion to right before checking the pub keys.

Reply:
With the `Assert` section, I think all assertions should be inside of it. Otherwise, we shouldn't use `Assert` section.